server/src/main/java/org/jboss/as/server/Main.java

```diff
@@ -392,6 +395,15 @@ public static ServerEnvironmentWrapper determineEnvironment(String[] args, Prope
                     } else {
                         gitBranch = arg.substring(idx + 1, arg.length());
                     }
+                } else if (arg.startsWith(CommandLineConstants.STREAM)) {
+                    String streamValue = parseValue(arg, CommandLineConstants.STREAM);
+                    try {
+                        stream = Stream.valueOf(streamValue.toUpperCase(Locale.ROOT));
+                    } catch (IllegalArgumentException e) {
+                        STDERR.println(ServerLogger.ROOT_LOGGER.failedToParseCommandLineEnum(arg, streamValue, EnumSet.allOf(Stream.class)));
+                        usage();
+                        return new ServerEnvironmentWrapper(ServerEnvironmentWrapper.ServerEnvironmentStatus.ERROR);
+                    }
```

Comment:
Practically the same changes should be done in org.jboss.as.host.controller.Main to initialize a HostControllerEnvironment with the stream passed as an argument.

We also have to verify if the ProcessController is able to pass these arguments to the command that launches the HostController. Is in class:

https://github.com/wildfly/wildfly-core/blob/08cb24dacd9615bb530b53067567c707a6184e16/process-controller/src/main/java/org/jboss/as/process/Main.java#L223-L247C2

The idea is when the process controller is launched with --stream, this argument value arrives at org.jboss.as.host.controller.Main to initialize the HostControllerEnvironment with it, and then, take it to initialize the Managed Servers launch command.

Reply:
Sounds good.  I'll add this today.

Reply:
OK - I've updated this.  Let me know if it looks ok now.

Reply:
Thank you, @pferraro LGTM